app/src/main/java/app/revanced/integrations/youtube/patches/VideoInformation.java

```diff
@@ -178,8 +197,31 @@ public static boolean seekTo(final long seekTime) {
             }
 
             Logger.printDebug(() -> "Seeking to " + adjustedSeekTime);
-            //noinspection DataFlowIssue
-            return (Boolean) seekMethod.invoke(playerControllerRef.get(), adjustedSeekTime);
+
+            try {
+                //noinspection DataFlowIssue
+                if ((Boolean) seekMethod.invoke(playerControllerRef.get(), adjustedSeekTime)) {
+                    return true;
+                }
+            } catch (Exception ex) {
+                Logger.printInfo(() -> "seekTo method call failed", ex);
+            }
+
+            // Try calling the seekTo method of the MDX player director (called when casting) if the player controller one failed.
+            // The difference has to be a different second mark in order to avoid infinite skip loops (as the Lounge API only supports seconds).
+            if (Math.abs(adjustedSeekTime/1000 - videoTime/1000) >= 1) {
+                try {
+                    //noinspection DataFlowIssue
+                    return (Boolean) mdxSeekMethod.invoke(mdxPlayerDirectorRef.get(), adjustedSeekTime);
+                } catch (Exception ex) {
+                    Logger.printInfo(() -> "seekTo (MDX) method call failed", ex);
+                    return false;
+                }
+            } else {
+                Logger.printDebug(() -> "Skipping seekTo for MDX because of a small relative value (" + (seekTime - videoTime) + "ms).");
+                return false;
+            }
```

Comment:
Uhm I think dividing the video time by 1000 makes your code more confusing, so here is my suggestion

```suggestion
            // Try calling the seekTo method of the MDX player director (called when casting) if the player controller one failed.
            // The difference has to be a different second (1000 miliseconds) mark in order to avoid infinite skip loops (as the Lounge API only supports seconds).
            if (Math.abs(adjustedSeekTime - videoTime) < 1000) {
                Logger.printDebug(() -> "Skipping seekTo for MDX because of a small relative value (" + (seekTime - videoTime) + "ms).");
                return false;
            }
            try {
                //noinspection DataFlowIssue
                return (Boolean) mdxSeekMethod.invoke(mdxPlayerDirectorRef.get(), adjustedSeekTime);
            } catch (Exception ex) {
                Logger.printInfo(() -> "seekTo (MDX) method call failed", ex);
                return false;
            }
```

Reply:
> > if (Math.abs(adjustedSeekTime - videoTime) < 1000)
> 
> That wont allow skipping to the next whole second, when the amount skipped is less than 1 second.
> 
> [#655 (comment)](https://github.com/ReVanced/revanced-integrations/pull/655#discussion_r1637260855)

I see, so how about this ones ?
```java
if ((adjustedSeekTime / 1000) == (videoTime / 1000)) {
```